src/frontend/src/optimizer/plan_node/mod.rs

```diff
@@ -677,6 +683,22 @@ impl Explain for PlanRef {
         serde_json::to_string_pretty(&PrettySerde(explain_ir))
             .expect("failed to serialize plan to json")
     }
+
+    /// Explain the plan node and return a xml string.
+    fn explain_to_xml(&self) -> String {
+        let plan = reorganize_elements_id(self.clone());
+        let explain_ir = plan.explain();
+        // serde_xml_rs::to_string(&PrettySerde(explain_ir))
```

Comment:
looks like there will be name collision on nested fields which will fail xml serialization? is there another way to do this without a new struct like `PrettySerde` or manually serialize?

Reply:
failed to serialize plan to xml: Writer { source: LastElementNameNotAvailable }
cc @kwannoel  for some insights

Reply:
I took a look at the `serde_xml_rs` library. Looks like it hasn't had activity for 2 years, and some bugs may exist. https://stackoverflow.com/questions/76526592/serde-xml-rs-lastelementnamenotavailable-when-serializing.

I suggest using `quick_xml` with the `serialize` feature. You may opt for a later version of `quick_xml`. I tried it locally -- It encounters recursion limit with `0.27` -- the version rw is using.